common/scala/src/main/scala/whisk/common/tracing/TraceUtil.scala

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright 2015-2016 IBM Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package whisk.common.tracing
+
+import java.util.concurrent.atomic.AtomicInteger
+import scala.collection.mutable.Map
+import akka.actor.ActorSystem
+import com.github.levkhomich.akka.tracing._
+import whisk.common.{LoggingMarkers, TransactionId}
+
+/**
+  * Trace Utility Class which provides all methods to trace a given service.
+  */
+object TraceUtil{
+
+  private lazy val trace: TracingExtensionImpl = traceVar;
+
+  private var traceVar: TracingExtensionImpl = null;
```

Comment:
Why use two members here? just `var trace: TracingExtensionImpl = null` should work.

Note that you would usually use `Option[TracingExtensionImpl]` to have NullPointer safe access later on.